pkg/cluster/clusterserviceprincipal.go

```diff
@@ -0,0 +1,163 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-09-01-preview/authorization"
+	"github.com/ghodss/yaml"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/rbac"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (m *manager) createOrUpdateClusterServicePrincipalRBAC(ctx context.Context) error {
+	resourceGroupID := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+	resourceGroup := stringutils.LastTokenByte(resourceGroupID, '/')
+
+	clusterSPObjectID, err := m.aad.GetServicePrincipalID(ctx)
+	if err != nil {
+		return err
+	}
+
+	roleAssignments, err := m.roleAssignments.ListForResourceGroup(ctx, resourceGroup, "")
+	if err != nil {
+		return err
+	}
+
+	// If we have Contributor RBAC role for Cluster SP on the resource group in question
+	// We are interested in Resource group scope only (inherited are returned too).
+	var toDelete []authorization.RoleAssignment
+	var found bool
+	for _, assignment := range roleAssignments {
+		// Contributor assignments only!
+		if *assignment.Scope == resourceGroupID && strings.HasSuffix(*assignment.RoleDefinitionID, rbac.RoleContributor) {
```

Comment:
I think `*assignment.Scope == resourceGroupID` needs to be case insensitive. I've seen both `resourcegroup` and `resourceGroup` in role assignment IDs. See below (different matching as it includes RG scope and lower instead of just RG, but it ignores case) https://github.com/Azure/ARO-RP/blob/64b7982737970fe60eae53b8f9921d42b248251a/pkg/util/cluster/cluster.go#L444-L447